redhat-developer/app-services-cli
# refactor: use the standard `os.UserConfigDir` API to get the config dir

Closes #626 

This PR achieves two things:

- Use the standard library for determining the config directory. We were implementing the same in the CLI with the exact same results.
- Change the default file name from `$HOME/.config/.rhoascli.json` to `$HOME/.config/rhoas/config.json`. Since the `.config` directory is already hidden, there is no longer a need to make the config hidden also. This change is backwards compatible by having a migration function in the code.
 
### Verification Steps

1. Run `rhoas whoami`
2. Check the location of your config file (it is different across OSs). It will have been renamed, but you will still be logged in.

### Type of change
